### PR TITLE
Adding app server messages 351-375

### PR DESCRIPTION
**Pull request by 1000TurquoisePogs**
Signed-off-by: 1000turquoisepogs <sgrady@rocketsoftware.com>

### CODE REVIEW

File: docs/troubleshoot/app-framework/appserver-error-codes.md [hunk 1]
```diff
@@ -2939,303 +2936,353 @@ The following error message codes may appear on the app-server log. Use the foll
...
 
-  TODO
+  Review the app-server schema https://github.com/zowe/zlux-app-server/blob/v2.x/master/schemas/app-server-config.json and your Zowe configuration file to identify and correct the missing properties within `components.app-server`.
+
 
 
+### ZWED0047E
 
-### ZWED0047E":"Proxy (%s:%s) setup failed.\nHost & Port for proxy destination are required but were missing.\nFor information on how to configure a proxy service, see the Zowe wiki on dataservices (https://github.com/zowe/zlux/wiki/ZLUX-Dataservices)
+ Proxy (_pluginid_:_servicename_) setup failed.\nHost & Port for proxy destination are required but were missing.\nFor information on how to configure a proxy service, see the Zowe wiki on dataservices (https://github.com/zowe/zlux/wiki/ZLUX-Dataservices)
```
Comment (DivergentEuropeans) [inline docs/troubleshoot/app-framework/appserver-error-codes.md:3025]: missed a "\n" here

Comment (DivergentEuropeans) [reply docs/troubleshoot/app-framework/appserver-error-codes.md:3025]: x2

File: docs/troubleshoot/app-framework/appserver-error-codes.md [hunk 2]
```diff
@@ -2939,303 +2936,353 @@ The following error message codes may appear on the app-server log. Use the foll
...
 
-  TODO
+  Confirm that the source plugin and service exist. Check the logs to see if there was something that caused the source service to fail loading. Contact the developers of either source or target plugin for more assistance if the cause is not clear.
 
 
 
-### ZWED0111E":"SEVERE: Exception occurred trying to generate object from input:%s
+### ZWED0111E
 
+ SEVERE: Exception occurred trying to generate object from input: _error_
```
Comment (DivergentEuropeans) [inline docs/troubleshoot/app-framework/appserver-error-codes.md:3109]: space here

File: docs/troubleshoot/app-framework/appserver-error-codes.md [hunk 3]
```diff
@@ -2939,303 +2936,353 @@ The following error message codes may appear on the app-server log. Use the foll
...
 
   **Reason:**
 
-  TODO
+  Within the list of _attributeKeys_, _attribute_ could not be found.
 
   **Action:**
 
-  TODO
+  Check the keystore configuration of the server such as in `zowe.certificate` or `components.app-server.node.https` to see if it is valid for Zowe. The SAF keyring Zowe was configured to use may be missing a key and certificate pair, or certificate authorities keychain.
```
Comment (DivergentEuropeans) [inline docs/troubleshoot/app-framework/appserver-error-codes.md:3201]: Could be good to link this: https://docs.zowe.org/stable/user-guide/configure-certificates-keyring/#create-a-self-signed-jceracfks-certificate

80% sure this is out of date, but we will update it

File: docs/troubleshoot/app-framework/appserver-error-codes.md [hunk 4]
```diff
@@ -2939,303 +2936,353 @@ The following error message codes may appear on the app-server log. Use the foll
...
 
   **Reason:**
 
-  TODO
+  The server configuration specified that the app-server should load keystore and truststore content from a SAF keyring, but the syntax in the configuration was incorrect, because _user_, _name_, or _label_ were not usable by the server.
 
   **Action:**
 
-  TODO
+  Check the `zowe.certificate` or `components.app-server.node.https` sections of Zowe configuration to see if there are entries that start with `safkeyring://` and verify that they are in the format of `safkeyring://USERNAME:RINGNAME&LABEL`. Older versions of zowe will require that there be 4 slashes, such as `safkeyring:////`. The `&LABEL` suffix is only needed for specifying certificate authorities and should be omitted in other sections, for example it is only needed within `zowe.certificate.pem.certificateAuthorities` or `components.app-server.node.https.certificateAuthorities`.
```
Comment (DivergentEuropeans) [inline docs/troubleshoot/app-framework/appserver-error-codes.md:3243]: Also a good place to link: https://docs.zowe.org/stable/user-guide/configure-certificates-keyring/#create-a-self-signed-jceracfks-certificate

File: docs/troubleshoot/app-framework/appserver-error-codes.md [hunk 5]
```diff
@@ -2939,303 +2936,353 @@ The following error message codes may appear on the app-server log. Use the foll
...
 
-### ZWED0152E":"Unable to locate server config instance location and INSTANCE_DIR environment variable does not exist.
+### ZWED0152E
+
+ Unable to locate server config instance location and INSTANCE_DIR environment variable does not exist.
 
   **Reason:**
 
-  TODO
+  Whilee installing a plugin, the server could not determine the location of the configuration dataservice's "instance" folder. Due to this, the plugin could not be completely installed.
```
Comment (DivergentEuropeans) [inline docs/troubleshoot/app-framework/appserver-error-codes.md:3281]: *While

**Review by DivergentEuropeans — CHANGES_REQUESTED** [review]

File: docs/troubleshoot/app-framework/appserver-error-codes.md (hunk 3 above)
Comment (1000TurquoisePogs) [reply docs/troubleshoot/app-framework/appserver-error-codes.md:3201]: maybe. im not totally comfortable with that doc. i expect user errors to result.

**Pull request merged**